server/service/src/ledger_fix/fixes/inventory_adjustment_to_balance.rs

```diff
@@ -0,0 +1,305 @@
+use chrono::TimeDelta;
+use repository::{
+    stock_line_ledger::{StockLineLedgerFilter, StockLineLedgerRepository, StockLineLedgerRow},
+    EqualFilter, InvoiceLineRow, InvoiceLineRowRepository, InvoiceLineType, InvoiceRow,
+    InvoiceRowRepository, InvoiceStatus, InvoiceType, ItemRowRepository, NameRowRepository,
+    NumberRowType, RepositoryError, StockLineRowRepository, StorageConnection,
+};
+use util::{
+    constants::{INVENTORY_ADJUSTMENT_NAME_CODE, SYSTEM_USER_ID},
+    uuid::uuid,
+};
+
+use crate::{
+    ledger_fix::{
+        fixes::{is_omsupply_uuid, LedgerFixError},
+        ledger_balance_summary, LedgerBalanceSummary,
+    },
+    number::next_number,
+};
+
+pub(crate) fn inventory_adjustment_to_balance(
+    connection: &StorageConnection,
+    operation_log: &mut String,
+    stock_line_id: &str,
+) -> Result<(), LedgerFixError> {
+    operation_log.push_str("Starting inventory_adjustment_to_balance\n");
+
+    let ledger_lines = StockLineLedgerRepository::new(connection).query_by_filter(
+        StockLineLedgerFilter::new().stock_line_id(EqualFilter::equal_to(stock_line_id)),
+    )?;
+
+    let balance_summary = ledger_balance_summary(connection, &ledger_lines, stock_line_id)?;
+    let LedgerBalanceSummary {
+        available,
+        total,
+        running_balance,
+        reserved_not_picked,
+        ..
+    } = balance_summary;
+
+    let should_adjust = available + reserved_not_picked == total
+        && total != running_balance
+        && !is_omsupply_uuid(stock_line_id);
+
+    if !should_adjust {
+        operation_log.push_str(&format!(
+            "Ledger does not match use case for inventory_adjustment_to_balance {:?}.\n",
+            balance_summary
+        ));
+        return Ok(());
+    }
+
+    let adjustment = running_balance * -1.0;
```

Comment:
This is not accurate, fixed upstream (should be adjusted based on total, to bring running balance to total